src/commands/settings/scan-members.ts

```diff
@@ -0,0 +1,79 @@
+import { CommandInteraction } from 'discord.js';
+import { WriteBatch } from '@google-cloud/firestore';
+
+// TODO: GDPR/CCPA compliance (applies to entire bot, tbh)
+// TODO: Could use cleanup, intentionally leaving until theres a usecase for this
+
+/** Collects guild member info into batches and processes them into Firebase */
+export async function scanMembers(
+  firestore: FirebaseFirestore.Firestore,
+  interaction: CommandInteraction
+) {
+  //Replying immediately allows command to run for longer than 3 seconds.
+  await interaction.reply({
+    content: 'Beginning user import',
+    ephemeral: true
+  });
+
+  const guild = interaction.guild;
+  if (guild === null) {
```

Comment:
Can guild possibly be `undefined`? I assume not?

Reply:
Discord.js types define it as `Guild | null`